app/lib/dns.server.ts

```diff
@@ -157,8 +168,23 @@ export const getChangeStatus = async (changeId: string) => {
   }
 };
 
-const isNameValid = (name: string) => {
-  return /^[a-z0-9-]+.[a-z0-9-]+.[a-z]+.[a-z]+\.?$/.test(name);
+/* Domain name rules
+1. Domain name pattern should be [name].[username].rootDomain.com
+2. Domain name can contain only alphanumerical characters, '-', and '_'
+3. Domain name should not start or end with -
+4. Domain name cannot contain multiple consecutive '-' or '_'
+5. Domain name can contain uppercase in UI but it is converted to lowercase before validation */
+export const isNameValid = (name: string, username: string) => {
+  const rootDomain = process.env.ROOT_DOMAIN!;
+  let subdomain = name.replace(`.${username}.${rootDomain}`, '');
```

Comment:
The problem with this solution is, that if replace does not find `.${username}.${rootDomain}`, it will do nothing (return the unchanged string). This is why I suggested that you do the .endsWith() first to confirm the correct structure.

Like, as is now, I think If I sent in 'localhost' as `name` (only that, nothing else), this would pass

Reply:
(Locahost is usually treated as a fully qualified domain name to my best knowledge)

Reply:
Absolutely valid suggestion. I will update.